File: command/java/SellStock.java
package Command;

// Nesnelere dönüştürdüğümüz her bir isteğe denk gelmektedir.
// Command (IOrderCommand) arayüzünü uygular.
// UML diyagramındaki ConcreteCommand sınıfına denk gelmektedir.
class SellStock implements IOrderCommand {
    private StockManager _stockManager;

    public SellStock(StockManager stockManager) {
        _stockManager = stockManager;
    }

    public void execute() {
        _stockManager.sell();
    }
}

File: command/typescript/sell-stock.ts
import {IOrderCommand} from "./order-command.interface";
import {StockManager} from "./stock-manager";

// Nesnelere dönüştürdüğümüz her bir isteğe denk gelmektedir.
// Command (IOrderCommand) arayüzünü uygular.
// UML diyagramındaki ConcreteCommand sınıfına denk gelmektedir.
export class SellStock implements IOrderCommand{
  constructor(private readonly stockManager: StockManager) {}

  public async execute(): Promise<void> {
    await this.stockManager.sell();
  }
}
